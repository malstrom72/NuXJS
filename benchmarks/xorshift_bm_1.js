function XorshiftPRNG2x32(seed0, seed1) {
	if (seed0 == null) seed0 = 123456789;
	if (seed1 == null) seed1 = 362436069;

	var px = seed0;
	var py = seed1;

	function next() {
		var t = px ^ (px << 10);
		px = py;
		py = py ^ (py >>> 13) ^ t ^ (t >>> 10);
	}

	this.nextInt32 = function() {
		next();
		return py >>> 0;
	};

	this.nextFloat = function() {
		next();
		return (py >>> 0) * 2.3283064365386962890625e-10 +
				(((px & 0xFFFFF800) >>> 0) * 5.42101086242752217003726400434970855712890625e-20);
	};

	this.nextInt = function(max) {
		var mask = max;
		mask |= mask >>> 1;
		mask |= mask >>> 2;
		mask |= mask >>> 4;
		mask |= mask >>> 8;
		mask |= mask >>> 16;
		mask |= mask >>> 32;
		
		var v;
		do {
			next();
			v = (py >>> 0) & mask;
		} while (v > max);

		return v;
	};

	this.getState = function() { return [ px, py ]; }
	this.setState = function(state) { px = state[0]; py = state[1]; }
	this.clone = function() { return new XorshiftPRNG2x32(px, py); };
}

var prng = new XorshiftPRNG2x32();
for (var j = 0; j < 100; ++j) {
	for (var i = 0; i < 10000; ++i) {
		prng.nextInt(12345678);
	}
	print(prng.nextInt(12345678));
}
