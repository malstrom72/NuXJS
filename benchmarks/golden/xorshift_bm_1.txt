1689566
6232109
9410238
11262603
307554
1141136
793826
4770016
10679222
5141659
6189890
2708406
6355278
8388287
5829526
1271544
11618177
7152155
4144820
7122172
9310979
9343907
6044892
6619462
6430933
7042436
7130817
3302765
1949020
536734
11444365
1838706
8992285
7474140
7335172
8526020
12311088
4284494
11804475
10762428
3593486
2611230
9332462
9783466
4120313
9791309
8325710
4156601
9258687
11725544
3422673
1406663
9545847
6601526
3177664
8462780
4550372
10817790
10302625
9267058
2291
10215563
7808491
4214485
588425
10167971
11374674
7659621
9164434
8074928
4746954
4526195
3080091
8165280
7093043
2878984
8624408
5397636
3025393
1899563
2598064
2107035
10653641
8633044
10414773
12103668
3150752
5962558
2286235
9524680
2844718
1958662
10214815
9956394
9428305
11712151
7551990
5530058
9317622
6853457
